Mathlib/CategoryTheory/WithTerminal/Basic.lean

```diff
@@ -447,6 +448,71 @@ def equivComma : (WithTerminal C ⥤ D) ≌ Comma (𝟭 (C ⥤ D)) (Functor.cons
 
 end
 
+open CategoryTheory.Limits CategoryTheory.Limits.WidePullbackShape
+
+instance subsingleton_hom {J : Type*} : Quiver.IsThin (WithTerminal (Discrete J)) := fun _ _ => by
+  constructor
+  intro a b
+  casesm* WithTerminal _, (_ : WithTerminal _) ⟶ (_ : WithTerminal _)
+  · exact congr_arg (ULift.up ∘ PLift.up) rfl
+  · rfl
+  · rfl
+
+private def widePullbackShapeEquivObj {J : Type*} :
```

Comment:
These definitions should not be made private, because the `simps` attribute below will makes these appear.